File: Tp1/Aquecimento Recursivo/Aquecimento.c
#include <stdio.h>

//Prototipos das funcoes
int quantasMaiusculas(char entrada[], int maiusculas, int tam);
int strComp(char entrada[]);


int main(){
    const int TAM = 1000;
    char entrada[TAM];

    fgets(entrada, TAM, stdin);

    while(!strComp(entrada)){
        printf("%d\n", quantasMaiusculas(entrada, 0, 0));
        fgets(entrada, TAM, stdin);
    }
    return 0;
}


int quantasMaiusculas(char *entrada, int maiusculas, int pos){  //Funcao de contar numero de maiusculas em uma frase

    if(entrada[pos] != '\0'){
        if(entrada[pos] >= 'A' && entrada[pos] <= 'Z'){
            maiusculas = quantasMaiusculas(entrada, maiusculas+1, pos+1);
        }
        else{
            maiusculas = quantasMaiusculas(entrada, maiusculas, pos+1);
        }
    }
    
    return maiusculas;
}


int strComp(char entrada[]){  //Funcao para finalizar o programa
    int ehFim = 1;

    if(entrada[0] != 'F' || entrada[1] != 'I' || entrada[2] != 'M'){
        ehFim = 0;
    }

    return ehFim;
}

File: TPs/Tp1/Aquecimento Recursivo/Aquecimento.java
import java.util.Scanner;

public class Aquecimento {
    public static void main(String[] args) {

        String frase = "";
        Scanner scanner = new Scanner(System.in);

        frase = scanner.nextLine();

        while (!StrComp(frase, "FIM")) {

            System.out.println(Maiusculas(frase, 0, 0));
            frase = scanner.nextLine();

        }



    }

    public static int Maiusculas(String frase, int maiusculas, int pos) {  //Funcao recursiva para contar numero de maiusculas em uma frase

        if (pos < frase.length()) {

            if (frase.charAt(pos) >= 'A' && frase.charAt(pos) <= 'Z') {
                maiusculas = Maiusculas(frase, maiusculas+1, pos+1);
            }
            else {
                maiusculas = Maiusculas(frase, maiusculas, pos+1);
            }

        }

        return maiusculas;

    }

    public static boolean StrComp (String a, String b) {  //Funcao para verificar a hora de parar o programa
        
        if (a.length() != b.length()) {
            return false;
        }

        for (int i = 0; i < a.length(); i++) {
            if (a.charAt(i) != b.charAt(i)) {
                return false;
            }
        }

        return true;
    }

}
